things/postgres/channels.go

```diff
@@ -149,22 +149,34 @@ func (cr channelRepository) RetrieveAll(ctx context.Context, owner string, offse
 		items = append(items, ch)
 	}
 
-	cq := ""
-	if name != "" {
-		cq = `AND LOWER(name) LIKE $2`
-	}
-
-	q = fmt.Sprintf(`SELECT COUNT(*) FROM channels WHERE owner = $1 %s;`, cq)
-
 	total := uint64(0)
+	cq := `SELECT COUNT(*) FROM channels WHERE owner = $1`
 	switch name {
 	case "":
-		if err := cr.db.GetContext(ctx, &total, q, owner); err != nil {
-			return things.ChannelsPage{}, err
+		switch metadata {
```

Comment:
this looks like lot of cases 
cannot this be done like here
https://github.com/mainflux/mainflux/blob/22f1e851cd334f30c33f314eff76f6af23088425/things/postgres/channels.go#L125 using getNameQuery and getMetadataQuery
is total different than number of rows we get from the query above?

Reply:
@mteodor The first one returns you one page after to apply filters. The second one, counts the total of Things with a certain filter in all pages.

Reply:
and this would not work
```cq := fmt.Sprintf(`SELECT COUNT(*) FROM things WHERE owner = $1 %s%s`, nq, mq)``` ?

Reply:
I don't see how here: https://godoc.org/github.com/jmoiron/sqlx#NamedQueryContext